staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -175,9 +162,54 @@ func (q *queue) dumpLocked(includeDetails bool) debug.QueueDump {
 
 	return debug.QueueDump{
 		NextDispatchR:     q.nextDispatchR.String(),
-		Requests:          digest,
-		ExecutingRequests: q.requestsExecuting,
+		Requests:          waitingDigest,
+		RequestsExecuting: executingDigest,
+		ExecutingRequests: q.requestsExecuting.Len(),
 		SeatsInUse:        q.seatsInUse,
 		QueueSum:          queueSum,
 	}
 }
+
+func (r *request) dump(includeDetails bool) debug.RequestDump {
+	ans := debug.RequestDump{
+		MatchedFlowSchema: r.fsName,
+		FlowDistinguisher: r.flowDistinguisher,
+		ArriveTime:        r.arrivalTime,
+		StartTime:         r.startTime,
+		WorkEstimate:      r.workEstimate.WorkEstimate,
+	}
+	if includeDetails {
+		userInfo, _ := genericrequest.UserFrom(r.ctx)
+		ans.UserName = userInfo.GetName()
+		requestInfo, ok := genericrequest.RequestInfoFrom(r.ctx)
+		if ok {
+			ans.RequestInfo = *requestInfo
+		}
+	}
+	return ans
+}
+
+func Curry2of2[T1, T2, Return any](f func(T1, T2) Return, v2 T2) func(T1) Return {
+	return func(v1 T1) Return {
+		return f(v1, v2)
+	}
+}
+
+func MapSetToSlice[EltInput comparable, EltOutput any](set sets.Set[EltInput], mapFn func(EltInput) EltOutput) []EltOutput {
+	ans := make([]EltOutput, 0, set.Len())
+	for elt := range set {
+		ans = append(ans, mapFn(elt))
+	}
+	return ans
+}
+
+func OrFuncOverSlice[Elt any](f func(Elt) bool) func([]Elt) bool {
```

Comment:
Given it's a function that is used only in the test, can we move it to that test file? I doubt we will ever remember about this function exists here...

Reply:
This is a form of map-reduce, I am sure I will remember that map-reduce exists and that I am tired of writing it over and over.

Reply:
Perhaps it should be

```go
func SliceMapReduce[Elt, Result any](mapFn func(Elt) Result, reduceFn func(Result,Result) Result) func([]Elt) Result {...{
```